core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -424,14 +429,17 @@ private void configureConnectors(SslContextFactory sslContextFactory) {
         SUPPORTED_URI_SCHEMES, "http");
 
     for (NamedURI listener : listeners) {
-      addConnectorForListener(httpConfiguration, httpConnectionFactory, listener, http2Enabled);
+      addConnectorForListener(httpConfiguration, httpConnectionFactory, listener,
+              http2Enabled, proxyProtocolEnabled);
     }
   }
 
   private void addConnectorForListener(HttpConfiguration httpConfiguration,
```

Comment:
Not really a comment to your change, but `addConnectorForListener` has become really hard to read. We should probably have an early return method that does the connection factories juggling and returns the set of relevant factories, and another one which creates a connector out of these.